### ISSUE TITLE
Export logs as json to Google Cloud Logging

### ISSUE THREAD

**Issue by ashish-b-choudhary-db** [open-telemetry/opentelemetry-collector-contrib#36845]
### Component(s)

exporter/googlecloud

### Is your feature request related to a problem? Please describe.

Currently the Google Cloud exporter sends the logs in an unstructured format under textPayload.

### Describe the solution you'd like

Provide option to send the logs in a structured manner as json with the right mappings.

### Describe alternatives you've considered

Going to try and look into transforming it ourselves

**Comment by dashpole** [comment]
The logging exporter will attempt to export json-structured logs if the value type is Map or Bytes, and is able to be converted to json: https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/8da0f42dab085c916987891419461d583a2aa96e/exporter/collector/logs.go#L503

Let me know if you need help formatting logs in that way.

**Comment by ashish-b-choudhary-db** [comment]
Yes please, would you have any suggestions on how to achieve the same.
Currently we're trying to use the https://opentelemetry.io/docs/zero-code/java/spring-boot-starter/ to send logs to a collector and then to Cloud Logging.
The native logs are structured while these logs are currently unstructured, we're trying to make the format same to have hybrid logs.

**Comment by dashpole** [comment]
You will want to use the `transform` processor.  You should be able to do something like this:

```yaml
transform:
  log_statements:
    - context: log
      statements:
      - set(attributes["body"], body)
      - set(body, attributes)
```

That moves the existing body into attributes, and then moves all attributes (which is a map) into the body

**Comment by ashish-b-choudhary-db** [comment]
@dashpole that did indeed work, however we didn't find the final result satisfactory.

So alternative we're exploring is to use log4j2 to write logs to a file in JSON having tracing information and hope to use that as the jsonPayLoad value.
However trying it out, the log value still went to textPayload.

Here's an example of app.log file content written using log4j2 and GcpLayout json template - 
`{"timestamp":"2025-01-03T04:22:56.262Z","severity":"INFO","message":"HV000001: Hibernate Validator 8.0.1.Final","logging.googleapis.com/sourceLocation":{"function":"org.hibernate.validator.internal.util.Version.<clinit>"},"logging.googleapis.com/insertId":"0","_exception":{"stackTrace":""},"_thread":"background-preinit","_logger":"org.hibernate.validator.internal.util.Version","logging.googleapis.com/trace_sampled":true}`

The value was read using filelog receiver and exported without transformation using googlecloud exporter:
```
receivers:
  filelog:
    include: [ /tmp/app.log ]
```

The final result in Cloud Logging looked like:
```
{
  "textPayload": "{\"timestamp\":\"2025-01-03T05:53:27.342Z\",\"severity\":\"INFO\",\"message\":\"HV000001: Hibernate Validator 8.0.1.Final\",\"logging.googleapis.com/sourceLocation\":{\"function\":\"org.hibernate.validator.internal.util.Version.<clinit>\"},\"logging.googleapis.com/insertId\":\"0\",\"_exception\":{\"stackTrace\":\"\"},\"_thread\":\"background-preinit\",\"_logger\":\"org.hibernate.validator.internal.util.Version\",\"logging.googleapis.com/trace_sampled\":true}",......
}
```

When usually the log is written to a container stdout and read by native GKE log agent, we get proper jsonPayload enriched with tracing information. We're looking to see if we achieve this using Otel collector too.
Any suggestions?
